package heron.proto.system;

option java_package = "com.twitter.heron.proto.system";
option java_outer_classname = "ExecutionEnvironment";

message AuroraJob {
  required string dc = 1;
  required string environ = 2;
  required string user = 3;
  required string jobname = 4;
}

message AuroraExecutionState {
  optional string release_username = 1;
  optional string release_tag = 2;
  // This is the user facing release version
  optional string release_version = 4;
  // This is the packer version
  optional string packer_version = 5;
  repeated AuroraJob jobs = 3;
}

message HeronReleaseState {
  optional string release_username = 1;
  optional string release_tag = 2;
  // This is the user facing release version
  optional string release_version = 3;
  // This is the uploader version
  optional string uploader_version = 4;
}

message ExecutionState {
  required string topology_name = 1;
  required string topology_id = 2;
  // The time the topology was submitted
  // Unix time
  optional int64 submission_time = 4;
  // The username who launched the topology
  optional string submission_user = 5;

  // Currently only aurora is supported.
  // TODO: Remove this completely.
  optional AuroraExecutionState aurora = 3;

  optional string dc = 6;
  optional string environ = 7;
  optional string role = 8;
  optional HeronReleaseState release_state = 9;
  optional string cluster = 10;
}
